File: quora/src/main/java/com/quoraclone/quora/entity/Question.java
package com.quoraclone.quora.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.Getter;
import lombok.Setter;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;

import java.util.Date;
import java.util.Set;

@Entity
@Table(name="question")
@Data
public class Question {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name="id")
    private Long id;

    @ManyToOne
    @JoinColumn(name="user_id", nullable = false)
    private User user;

    @Column(name="text")
    private String text;

    @OneToMany(cascade = CascadeType.ALL, mappedBy = "question")
    private Set<Answer> answers;

    // Created timestamp
    @Temporal(TemporalType.TIMESTAMP)
    @CreatedDate
    private Date created;

    // Last modified timestamp
    @Temporal(TemporalType.TIMESTAMP)
    @LastModifiedDate
    private Date last_updated;
}

File: frontend-01/quora/src/app/common/question.ts
export class Answer{
    id: number;
    text: string;
    username: string;
    owner: boolean;

    constructor(a: Answer){
        this.id = a.id;
        this.text = a.text;
        this.username = a.username;
        this.owner = a.owner;
    }
}

export class Question {
    id : number;
    text: String;
    userId: number;
    username: String;
    answers: Answer[];
    sumVotes: number;
    owner: boolean;
    created: Date;

    constructor(q: Question){
        this.id=q.id;
        this.text = q.text;
        this.userId = q.userId;
        this.username = q.username;

        this.answers = q.answers;
        this.sumVotes = q.sumVotes;
        this.owner = q.owner;
        this.created = q.created;
    }
}
